flybywire-aircraft-a320-neo/html_ui/Pages/A32NX_Core/A32NX_Transition.js

```diff
@@ -0,0 +1,78 @@
+class A32NX_Transition {
+    init() {
+        console.log('A32NX_TransitionAltitude init');
+
+        /*
+        *   This file build manually by hand & some python crawling on each airport charts from their EAIP.
+        *   So it may contain error. Please fix errors if you find it.
+        */
+        this.transitionSource;
+        this.loadJSON('/Database/transitionAltitude.json');
+
+        // Initial Airport
+        this.originAirport = "";
+        this.destinationAirport = "";
+
+        // Saved Airport
+        this.savedOriginAirport = "";
+        this.savedDestinationAirport = "";
+    }
+
+    update() {
+        if (this.remote) {
+            this.trySetTA("Origin", this.originAirport);
+            this.trySetTA("Destination", this.destinationAirport);
+            this.remote = false;
+        }
+        this.TAUpdate();
+    }
+
+    async loadJSON(location) {
+        const data = await fetch(location);
+        const text = await data.text();
+        this.transitionSource = await JSON.parse(text);
+        console.log("JSON IMPORT COMPLETE");
+    }
+
+    search(icao) {
+        const result = this.transitionSource.filter(data => {
+            if (data.icao === icao) {
+                return icao;
+            }
+        });
+        return result;
+    }
+
+    TAUpdate() {
+        if (this.originAirport !== "" && this.destinationAirport !== "") {
+            if (this.originAirport !== this.savedOriginAirport) {
+                this.trySetTA("Origin", this.originAirport);
+                this.savedOriginAirport = this.originAirport;
+            }
+            if (this.destinationAirport !== this.savedDestinationAirport) {
+                this.trySetTA("Destination", this.destinationAirport);
+                this.savedDestinationAirport = this.destinationAirport;
+            }
+        }
+    }
+
+    trySetTA(mode, icao) {
+        if (mode === "Origin") {
+            const varName = "L:AIRLINER_TRANS_ALT";
+            const ta = this.search(icao)[0].transitionAltitude;
+            this.setTransitonAltitude(varName, ta);
+        } else if (mode === "Destination") {
+            const varName = "L:AIRLINER_APPR_TRANS_ALT";
+            const ta = this.search(icao)[0].transitionLevel;
+            this.setTransitonAltitude(varName, ta);
+        }
+    }
+
+    setTransitonAltitude(varName, ta) {
+        if (ta !== null) {
+            SimVar.SetSimVarValue(varName, "Number", ta);
+        } else {
+            SimVar.SetSimVarValue(varName, "Number", 18000);
```

Comment:
```suggestion
            SimVar.SetSimVarValue(varName, "Number", 18000);
```

should I remove this?

msfs default TA / TL is 18000 as US.
but IRL there is no default TA for aircraft on each airport

Reply:
Each airport has it's own TA, since you've added a database for these, why do we need a distinct default?

Reply:
I was added that if failed to search or there is no TA value, then add default to 18000

so... good to remove i think?

Reply:
I would say to keep the value there as you did mention that there is potential for error, especially with such a large amount of data. Good precautions 👍